ServiceWebsite/ServiceWebsite.Services/HearingsService.cs

```diff
@@ -1,28 +1,53 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Linq;
+using System.Net;
 using System.Threading.Tasks;
+using ServiceWebsite.BookingsAPI.Client;
 using ServiceWebsite.Common;
 using ServiceWebsite.Domain;
 
 namespace ServiceWebsite.Services
 {
     public class HearingsService : IHearingsService
     {
-        public async Task<Hearing[]> GetHearingsFor(string userEmail)
+        private readonly IBookingsApiClient _bookingsApiClient;
+
+        public HearingsService(IBookingsApiClient bookingsApiClient)
         {
-           
-            List<Hearing> hearings = new List<Hearing>();
-            for(int i=1;i<=5;i++)
+            _bookingsApiClient = bookingsApiClient;
+        }
+
+        public async Task<Hearing> GetHearingFor(string username, Guid id)
+        {
+            try
+            {
+                var hearingResponse = await _bookingsApiClient.GetHearingDetailsByIdAsync(id);
+                if (!HearingContainsParticipant(hearingResponse, username))
+                {
+                    throw new UnauthorizedAccessException("User is not participant of hearing: " + id);
+                }
+                return Map(hearingResponse);
+            }
+            catch (BookingsApiException e)
             {
+                if (e.StatusCode == (int) HttpStatusCode.NotFound)
+                {
+                    throw new NotFoundException("Could not find hearing with id: " + id);
+                }
 
-                hearings.Add(new Hearing(1,"Case "+i,"Case "+i));
+                throw;
             }
-            return await Task.FromResult(hearings.ToArray());
-           
         }
 
-        private DateTime Today => DateTime.UtcNow.Date;
-        private DateTime NextYear => DateTime.Today.AddYears(1).Date;
+        private static bool HearingContainsParticipant(HearingDetailsResponse response, string username)
+        {
+            return response.Participants.Any(p => p.Username.Equals(username, StringComparison.OrdinalIgnoreCase));
+        }
+        
+        private static Hearing Map(HearingDetailsResponse response)
+        {
+            var hearingCase = response.Cases.FirstOrDefault(c => c.Is_lead_case.GetValueOrDefault(true)) ?? response.Cases.First();
```

Comment:
Will throw on response.Scheduled_date_time.Value if value null

Reply:
never be null, it's mandatory